lib/taskana-core-test/src/test/java/acceptance/task/complete/CancelTaskAccTest.java

```diff
@@ -0,0 +1,195 @@
+
+@TaskanaIntegrationTest
+class CancelTaskAccTest {
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject UserService userService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+
+  @WithAccessId(user = "admin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-2")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    defaultObjectReference = defaultTestObjectReference().build();
+  }
+
+  @WithAccessId(user = "admin")
+  @Test
+  void should_CancelReadyTask() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .state(TaskState.READY)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    taskService.cancelTask(task.getId());
+
+    Task readTask = taskService.getTask(task.getId());
+    assertThat(readTask.getState()).isEqualTo(TaskState.CANCELLED);
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_CancelTask_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
+      throws Exception {
+    Task taskToCancel =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .state(TaskState.CLAIMED)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    Task cancelledTask = taskService.cancelTask(taskToCancel.getId());
+
+    assertThat(cancelledTask.getState()).isEqualTo(TaskState.CANCELLED);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_CancelClaimedTask() throws Exception {
+    Task claimedTask =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .state(TaskState.CLAIMED)
+            .primaryObjRef(DefaultTestEntities.defaultTestObjectReference().build())
+            .buildAndStore(taskService);
+
+    taskService.cancelTask(claimedTask.getId());
+
+    Task readTask = taskService.getTask(claimedTask.getId());
```

Comment:
Please remove the line `Task readTask = taskService.getTask(claimedTask.getId());` and read the return value of cancelTask instead